install/kubernetes/helm/istio/charts/prometheus/templates/operator.yaml

```diff
@@ -0,0 +1,331 @@
+{{- if .Values.operator.createServiceMonitors -}}
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: istio-mesh-monitor
+  namespace: istio-system
+  labels:
+    monitoring: istio-mesh
+spec:
+  selector:
+    matchExpressions:
+      - {key: istio, operator: In, values: [mixer]}
+  namespaceSelector:
+    matchNames:
+      - istio-system
+  endpoints:
+  - port: prometheus
+    interval: {{ .Values.scrapeInterval }}
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: istio-component-monitor
+  namespace: istio-system
+  labels:
+    monitoring: istio-components
+spec:
+  selector:
+    matchExpressions:
+      - {key: istio, operator: In, values: [mixer,pilot,galley,citadel]}
+  namespaceSelector:
+    matchNames:
+      - istio-system
+  endpoints:
+  - port: http-monitoring 
+    interval: {{ .Values.scrapeInterval }}
+---
+apiVersion: monitoring.coreos.com/v1
+kind: ServiceMonitor
+metadata:
+  name: envoy-stats-monitor
+  namespace: istio-system
+  labels:
+    monitoring: istio-proxies
+spec:
+  selector:
+    matchExpressions:
+      - {key: istio-prometheus-ignore, operator: DoesNotExist}
+  namespaceSelector:
+    any: true
+  jobLabel: envoy-stats
+  endpoints:
+  - path: /stats/prometheus
+    targetPort: 15090
+    interval: {{ .Values.scrapeInterval }}
+    relabelings:
+    - sourceLabels: [__meta_kubernetes_pod_container_port_name]
+      action: keep
+      regex: '.*-envoy-prom'
+    - action: labelmap
+      regex: "__meta_kubernetes_pod_label_(.+)"
+    - sourceLabels: [__meta_kubernetes_namespace]
+      action: replace
+      targetLabel: namespace
+    - sourceLabels: [__meta_kubernetes_pod_name]
+      action: replace
+      targetLabel: pod_name
+    metricRelabelings:
+    - sourceLabels: [ cluster_name ]
+      regex: '(outbound|inbound|prometheus_stats).*'
+      action: drop
+    - sourceLabels: [ tcp_prefix ]
+      regex: '(outbound|inbound|prometheus_stats).*'
+      action: drop
+    - sourceLabels: [ listener_address ]
+      regex: '(.+)'
+      action: drop
+    - sourceLabels: [ http_conn_manager_listener_prefix ]
+      regex: '(.+)'
+      action: drop
+    - sourceLabels: [ http_conn_manager_prefix ]
```

Comment:
Since we restrict the set of metrics at source by using inclusion list, many of these are not needed now.

Reply:
That's true. We left them initially to double-cover ourselves, but this probably is a good opportunity to trim the fat from the relabeling rules.